aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,368 @@
+        )
+        dialogue_reference = (
+            dialogue_reference
+            if dialogue_reference is not None
+            else dialogue.dialogue_label.dialogue_reference
+        )
+        message_id = (
+            message_id
+            if message_id is not None
+            else dialogue.last_message.message_id + 1
+        )
+        target = target if target is not None else dialogue.last_message.message_id
+        to = to if to is not None else self.skill.skill_context.agent_address
+        sender = (
+            sender
+            if sender is not None
+            else dialogue.dialogue_label.dialogue_opponent_addr
+        )
+
+        incoming_message = self.build_incoming_message(
+            message_type=message_type,
+            performative=performative,
+            dialogue_reference=dialogue_reference,
+            message_id=message_id,
+            target=target,
+            to=to,
+            sender=sender,
+            **kwargs,
+        )
+
+        return incoming_message
+
+    @staticmethod
+    def _provide_unspecified_fields(
+        message: DialogueMessage, last_incoming: Optional[bool], message_id: int
+    ) -> Tuple[bool, int]:
+        default_is_incoming = not last_incoming
+        is_incoming = default_is_incoming if message[2] is None else message[2]
+
+        default_target = message_id - 1
+        target = default_target if message[3] is None else message[3]
+        return is_incoming, target
+
+    @staticmethod
+    def _non_initial_incoming_message_dialogue_reference(
+        dialogue: Dialogue,
+    ) -> Tuple[str, str]:
+        dialogue_reference = (
+            dialogue.dialogue_label.dialogue_reference[0],
+            Dialogues._generate_dialogue_nonce()  # pylint: disable=protected-access
+            if dialogue.dialogue_label.dialogue_reference[1]
+            == Dialogue.UNASSIGNED_DIALOGUE_REFERENCE
+            else dialogue.dialogue_label.dialogue_reference[1],
+        )
+        return dialogue_reference
+
+    def prepare_dialogue(
+        self,
+        dialogues: Dialogues,
+        messages: Tuple[DialogueMessage, ...],
+        counterparty: Address = "counterparty",
+    ) -> Dialogue:
+        """
+        Quickly create a dialogue.
+
+        The 'messages' argument is a tuple of DialogueMessages.
+        For every DialogueMessage (performative, contents, is_incoming, target):
+            - if 'is_incoming' is not provided: for the first message it is assumed False (outgoing),
+            for any other message, it is the opposite of the one preceding it.
+            - if 'target' is not provided: for the first message it is assumed 0,
+            for any other message, it is the index of the message before it in the tuple of messages + 1.
+
+        :param dialogues: a dialogues class
+        :param counterparty: the message_id
+        :param messages: the dialogue_reference
+
+        :return: the created incoming message
+        """
```

Comment:
First we need to check the messages list is non-zero length.

Reply:
Done